core/os/memory.h

```diff
@@ -51,6 +47,10 @@ class Memory {
 	static SafeNumeric<uint64_t> alloc_count;
 
 public:
+	static constexpr size_t SIZE_OFFSET = 0;
```

Comment:
Might want to add the explanation of the offsets here (like they are in CowData), maybe even instead of in the cpp file where they are defined, as the maths here is a bit dense without an explanation

Reply:
Moved comment to the header.